packages/http-client/src/typekit/kits/client.ts

```diff
@@ -169,11 +171,35 @@ defineKit<TypekitExtension>({
         if (isTemplateDeclarationOrInstance(clientOperation)) {
           continue;
         }
-
         operations.push(clientOperation);
       }
 
-      const httpOperations = operations.map((o) => this.httpOperation.get(o));
+      const httpOperations = operations.map((o) => {
+        const httpOperation = this.httpOperation.get(o);
+
+        const streamMetadataBody = getStreamMetadata(this.program, httpOperation.parameters);
+        const bytes = ignoreDiagnostics(this.program.resolveTypeReference("TypeSpec.bytes"));
+        httpOperation.parameters.properties.map((param) => {
+          if (
+            param.kind === "body" &&
+            streamMetadataBody?.bodyType === param.property.type &&
+            bytes
+          ) {
+            param.property.type = bytes;
```

Comment:
I don't think we should mutate the types this way. If we need to mutate we should do it through the mutators instead so that we don't impact the typegraph